src/nodes.jl

```diff
@@ -0,0 +1,95 @@
+using SparseArrays
+
+function node_pointers(nodes::AbstractMatrix{SVMNode})
+    nrows, ncols = size(nodes)
+    pointers = Array{Ptr{SVMNode}}(undef, ncols)
+
+    for i in 1:ncols
+        pointers[i] = pointer(nodes, (i - 1) * nrows + 1)
+    end
+
+    return pointers
+end
+
+function fill_nodes!(nodes::AbstractMatrix{SVMNode},
+                     X::AbstractMatrix{<:Real})
+    nrows, ncols = size(X)
+
+    # The last row will be filled with terminal nodes as required by
+    # SVMLIB
+    @assert size(nodes, 1) == nrows + 1
+    @assert size(nodes, 2) == ncols
+
+    for i in 1:ncols
+        for j in 1:nrows
+            nodes[j, i] = SVMNode(j, X[j, i])
+        end
+        nodes[end, i] = SVMNode(-1, NaN)
```

Comment:
Could this terminal node be shared?
maybe create a `const` terminal node at global scope.

Reply:
Seems safe & the tests pass.